tests/test_report.py

```diff
@@ -677,3 +677,66 @@ def test_show_empty_show_missing(self):
         # headings and data should be unaltered
         self.assertEqual(self.headings, headings)
         self.assertEqual(self.sample_data, out_data)
+
+
+class TestMultiReport(unittest.TestCase):
+    """Tests for the MultiReport class"""
+    def setUp(self):
+        self.multireport = MultiReport()
+        self.reports = {
+            'bread_recipe': [
+                ['flour', '500 g'],
+                ['water', '300 mL'],
+                ['yeast', '2 g']
+            ],
+            'variables': [
+                ['foo'],
+                ['bar']
+            ],
+        }
+        self.recipe_headings = ['ingredient', 'amount']
+        self.variables_headings = ['name']
+        self.recipe_report = self.multireport.add_report('Bread Recipe', self.recipe_headings)
+        self.variables_report = self.multireport.add_report('Variables', self.variables_headings)
+
+        self.recipe_report.add_rows(self.reports['bread_recipe'])
+        self.variables_report.add_rows(self.reports['variables'])
+
+    def test_adding_multiple_reports(self):
+        """Test that reports are added with the correct headings"""
+        for idx, headings in enumerate([self.recipe_headings, self.variables_headings]):
+            self.assertEqual(self.multireport.reports[idx].headings, headings)
+
+    def test_pretty_printing_multireport(self):
+        """Test pretty-printing a MultiReport"""
+        pretty_report = self.multireport.get_formatted_report('pretty')
+        self.assertIn(self.recipe_report.get_formatted_report('pretty'), pretty_report)
+        self.assertIn(self.variables_report.get_formatted_report('pretty'), pretty_report)
+
+    def test_multireport_with_multiple_formats(self):
+        """Test that a MultiReport with multiple formats in use defaults to pretty format"""
+        for recipe_format, variables_format in combinations(['pretty', 'json', 'yaml'], 2):
```

Comment:
I think you might have meant to use `combinations_with_replacement` if you are expecting to test the case when they are actually equal below.